File: jar/src/main/java/jline/api/measures/Ms_condentropy.java
package jline.api.measures;

import static jline.api.measures.Ms_entropy.ms_entropy;
import static jline.api.measures.Ms_jointentropy.ms_jointentropy;

import jline.util.matrix.Matrix;

public final class Ms_condentropy {
    private Ms_condentropy() {}

    /**
     * Compute conditional entropy z=H(x|y) of two discrete variables x and y.
     *
     * @param x first matrix
     * @param y second matrix of the same length
     * @return conditional entropy z=H(x|y)
     */
    public static double ms_condentropy(Matrix x, Matrix y) {
        if (x.length() != y.length()) {
            throw new IllegalArgumentException("Input matrices must have the same length");
        }

        if (x.isEmpty()) {
            return 0.0;
        }

        // H(x|y) = H(x,y) - H(y)
        double hxy = ms_jointentropy(x, y);
        double hy = ms_entropy(y);

        double condEntropy = hxy - hy;
        return Math.max(0.0, condEntropy);
    }
}

File: jar/src/main/kotlin/jline/api/measures/Ms_condentropy.kt
package jline.api.measures

import jline.util.matrix.Matrix
import kotlin.math.max

/**
 * Compute conditional entropy z=H(x|y) of two discrete variables x and y.
 * 
 * @param x first matrix
 * @param y second matrix of the same length
 * @return conditional entropy z=H(x|y)
 */
fun ms_condentropy(x: Matrix, y: Matrix): Double {
    require(x.length() == y.length()) { "Input matrices must have the same length" }
    
    if (x.isEmpty) return 0.0
    
    // H(x|y) = H(x,y) - H(y)
    val hxy = ms_jointentropy(x, y)
    val hy = ms_entropy(y)
    
    val condEntropy = hxy - hy
    return max(0.0, condEntropy)
}